ruma-identifiers/src/event_id.rs

```diff
@@ -55,18 +55,15 @@ impl<T> EventId<T> {
     /// parsed as a valid host.
     #[cfg(feature = "rand")]
     #[cfg_attr(docsrs, doc(cfg(feature = "rand")))]
-    pub fn new(server_name: &str) -> Result<Self, Error>
+    pub fn new(server_name: &ServerName<&str>) -> Self
```

Comment:
If we keep `ServerName` as it is, this should be

```suggestion
    pub fn new(server_name: ServerNameRef<'_>) -> Self
```

And even if we change it to a DST, we may want to go with that instead of `&ServerName` for consistency with the other identifier types. Using a reference to one of the `Id<&str>` types doesn't make sense – they're very small and implement `Copy`.